src/renderer/ssf-api.ts

```diff
@@ -263,6 +264,33 @@ export class SSFApi {
         }
     }
 
+    /**
+     * Allows JS to register a log retriever that can be used by the
+     * electron main process to retrieve current logs.
+     */
+    public registerLogRetriever(collectLogs: (logName: string) => void, logName: string): void {
+        if (typeof collectLogs === 'function') {
+
+            local.collectLogsCallback = collectLogs;
```

Comment:
Will this be overwritten if there are multiple log providers?

Reply:
Ah, right, this should probably be a list?